akka-remote/src/main/scala/akka/remote/artery/ArteryTransport.scala

```diff
@@ -563,17 +553,27 @@ private[remote] class ArteryTransport(_system: ExtendedActorSystem, _provider: R
               case ActorRefCompressionAdvertisement(from, table) ⇒
                 log.debug("Incoming ActorRef compression advertisement from [{}], table: [{}]", from, table)
                 val a = association(from.address)
-                a.outboundCompression.applyActorRefCompressionTable(table)
-                a.sendControl(ActorRefCompressionAdvertisementAck(localAddress, table.version))
-                system.eventStream.publish(Events.ReceivedActorRefCompressionTable(from, table))
+                // make sure uid is same for active association
+                if (a.associationState.uniqueRemoteAddressValue().exists(_ == from)) {
```

Comment:
I think I nailed this thing. See 0983568
Now clearing of outbound compression tables is part of the handshake process.